sygnal/apnspushkin.py

```diff
@@ -313,15 +313,11 @@ async def _dispatch_notification_unlimited(
                     span_parent.log_kv(
                         {"event": "temporary_fail", "retrying_in": retry_delay}
                     )
+                    await twisted_sleep(
+                        retry_delay, twisted_reactor=self.sygnal.reactor
+                    )
```

Comment:
```suggestion
                    if retry_number < self.MAX_TRIES - 1:
                        await twisted_sleep(
                            retry_delay, twisted_reactor=self.sygnal.reactor
                        )
```
subtle, but this stops it from sleeping when we've already given up.

(`try sleep try sleep try fail` rather than `try sleep try sleep try sleep fail`)